spec/classes/puppet_agent_spec.rb

```diff
@@ -110,6 +110,21 @@ def global_facts(facts, os)
         end
 
         [{}, {:service_names => []}].each do |params|
+          context "puppet_agent class with install_options" do
+            let(:params) { global_params.merge(
+              {:install_options => ['OPTION1=value1','OPTION2=value2'],})
+            }
+
+            it { is_expected.to compile.with_all_deps }
```

Comment:
Still a work in progress.  Running spec test on Windows is difficult for this module.